Fix Windows XMOS handling in audio control utils.

brainwavecoder9 opened the pull request:
Added Windows compatibility for XMOS XVF3800 USB control by handling platform-specific kernel driver operations.

Closes #442

brainwavecoder9 commented:
FYI this fixes Windows, but I think updated udev rules are still necessary for Linux (the ones [suggested](https://discord.com/channels/519098054377340948/1428484261802934312/1442818761668427856) give insufficient permission issue, also tried ATTRS)

CarolinePascal commented:
Hi @brainwavecoder9 !

First, I'd like to thank you (again) for hunting down these nasty bugs on Windows 🫶 

It's a very good catch for the missing `backend = ...`, but I don't see why the configuration change is needed. Could you describe the bug you had before adding the fix ? I'll try to find a Linux machine to replicate it !

Best,

Caroline.

brainwavecoder9 commented:
I think this is still a necessary improvement for Windows, but yes removing any need for users to interact with this is actually the ideal solution.

I don't believe I have access to the firmware source, but I think the reason we need to REBOOT is because these modules are failing to initialize on cold boot. I see errors and timeouts related to module not ready or module not responding properly for:

- AEC (Acoustic Echo Cancellation) - Resource ID 33
- AUDIO_MGR (Audio Manager) - Resource ID 35
- PP (Post-Processing) - Resource ID 17
- I2S (Audio Interface) - Resource ID 35

brainwavecoder9 commented:
Caroline's idea of removing the need for users to do this entirely is for sure the right way to go, but FYI that I did confirm on another Linux system that the udev rules are necessary. 

FYI what worked for me: 
```
	sudo nano /etc/udev/rules.d/99-reachy.rules
		# For 2.1.2 & 2.1.3 firmware
		SUBSYSTEM=="usb", ATTR{idVendor}=="38fb", ATTR{idProduct}=="1001", MODE="0666", GROUP="plugdev"
		SUBSYSTEM=="usb", ATTR{idVendor}=="2886", ATTR{idProduct}=="001a", MODE="0666", GROUP="plugdev"

	# refresh
	sudo udevadm control --reload-rules
	sudo udevadm trigger
	
	# add current user to have perms 
	sudo usermod -aG plugdev $USER
```

brainwavecoder9 commented:
Ready for pull:
- Confirmed success on Windows, and that util doesn't mind if daemon is running
- Enabled original intent of firmware check, clarified logic 
- Removed kernel ops in draft
- Cleaned up messaging

FabienDanieau commented:
Rules are necessary on Linux indeed. We can add them in the firmware folder and a install script. This could be useful for some beta testers. The REBOOT is still necessary although the bug should be very less frequent.
Since this is a Windows oriented issue here we can open another one dedicated to the linux udev rules. As you prefer.
